### PR TITLE
#159072345  Reject event invite 

### PR DESCRIPTION
**Pull request by andela-mmakinde**
#### What Does This PR Do?
 - Allows users to reject invite to an event
- Allows users to query the user model
- Allows users to create categories from the graphQL endpoint

#### Description Of Task To Be Completed
- After directing users to an event page(through slack, invite link, etc.), a user can either accept an invitation to attend the event or decline the invite. This PR allows the user to decline an event invite. 

#### Any Background Context You Want To Provide?
 - This PR has many file changes as because I had issues while querying the user endpoint from qraphQL. To fix this I had to create a user node on qraphQL

#### How can this be manually tested?
 - After the necessary user, category and event tables have been populated appropriately.
- Create a mutation of this form 
            `declineEvent(input: {eventId: "RXZlbnROb2RlOjI="}) {`
               `clientMutationId`
                `declinedEvent {`
                   `  id`
                    ` event {`
                      ` id`
                        `socialEvent {`
                            `id`
                        `}`
                    `}`
                    `user {`
                        `id`
                    `}`
                `}`
            `}`
#### What are the relevant pivotal tracker stories?
#159072345

### CODE REVIEW

File: server/api/models.py [hunk 1]
```diff
@@ -332,3 +332,18 @@ def save(self, *args, **kwargs):
...
+    user = models.ForeignKey(AndelaUserProfile, on_delete=models.CASCADE)
+    event = models.ForeignKey(Event, on_delete=models.CASCADE)
+
+    class Meta:
+        ordering = ('-created_at',)
+        unique_together = ('user', 'event')
+
+    def __str__(self):
+        return "@{} declined attending event {}" .format(
+            self.user.slack_name, self.event.title)
```
Comment (tomipaul) [inline server/api/models.py]: @andela-mmakinde good stuff!
Just a side thought, is there a way we could have done this using the already existing Attend model without breaking the interface? Say introduce a field that stores acceptance or declination.

Comment (makindemayowa) [reply server/api/models.py]: @tomipaul Yes it's very much possible. But personally, I'll rather have the concerns separated. And again if I were to go that route, then there be changes in the current way the attendance is structured. 
I'll want the team to weigh in on this though.

Comment (tosmak16) [reply server/api/models.py]: Well done! @makindemayowa I support we discuss this during our week sync tomorrow.

**Comment by parseendavid** [comment]
Good work @andela-mmakinde aside from the discussion raised above by @tomipaul and @tosmak16 , LGTM

**Comment by andela-mmakinde** [comment]
closed in favour of
https://github.com/AndelaOSP/Andela-Socials/pull/73
https://github.com/AndelaOSP/Andela-Socials/pull/67
https://github.com/AndelaOSP/Andela-Socials/pull/66

**Pull request closed without merging**